File: 0950-reveal-cards-in-increasing-order/0950-reveal-cards-in-increasing-order.java
class Solution {
    public int[] deckRevealedIncreasing(int[] deck) {
        Arrays.sort(deck);

        int n = deck.length;
        Deque<Integer> que = new ArrayDeque<>();
        int[] result = new int[n];

        for (int i = 0; i < n; i++) {
            que.addLast(i);
        }

        for (int i = 0; i < n; i++) {
            result[que.pollFirst()] = deck[i];
            if (!que.isEmpty()) {
                que.addLast(que.pollFirst());
            }
        }

        return result;
    }
}

File: 0950-reveal-cards-in-increasing-order/0950-reveal-cards-in-increasing-order.py
class Solution:
    def deckRevealedIncreasing(self, deck: List[int]) -> List[int]:
        deck.sort()
        que = deque(range(len(deck)))
        result = [0 for i in range(len(deck))]

        for i in range(len(deck)):
            result[que.popleft()] = deck[i]
            if que:
                que.append(que.popleft())

        return  result